### PR TITLE
[Component framework]: Reduce the usage of demultiplexerInstance instance

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR reduces the usage of `demultiplexerInstance`. This is a first step in order to remove `demultiplexerInstance`.
This PR introduces `GetSenderManager()` and  injects it in the `Configure` method of a `Check` in order to let the checks send metrics without using `demultiplexerInstance`.

In a follow-up PR, `SenderManager` will become a component.

**This is strongly recommended to review commit per commit** 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]
👍 for ebpf-platform owned files

**Review by pippenpaddleopsicopolis — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned files !

**Review by clarkb7 — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]
LGTM for ASC files

**Review by modernplumbing — APPROVED** [review]

**Review by just-chillin — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]
Looks fine for APM

**Review by davidor — APPROVED** [review]
:+1: for files owned by container-integrations

**Review by mfpierre — APPROVED** [review]
LGTM for owned file (orchestrator check) 👍

File: pkg/aggregator/mocksender/mocksender.go [hunk 1]
```diff
@@ -3,8 +3,6 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build test
-
```
Comment (vickenty) [inline pkg/aggregator/mocksender/mocksender.go]: Is this intentional? We shouldn't need a mock in the final binary.

**Review by vickenty — APPROVED** [review]
Small nit, otherwise LGTM

**Review by paulcacheux — APPROVED** [review]

**Review by jmeunier28 — APPROVED** [review]

**Review by alexandre-normand — APPROVED** [review]

**Pull request merged**